packages/stylex/src/StyleXSheet.js

```diff
@@ -280,54 +289,123 @@ export class StyleXSheet {
   /**
    * Insert a rule into the stylesheet.
    */
-  insert(rawLTRRule: string, priority: number, rawRTLRule: ?string) {
-    // Inject the stylesheet if it hasn't already been
-    if (this.injected === false) {
+  insert(
+    rawLTRRule: string,
+    priority: number,
+    rawRTLRule: ?string,
+    metadata?: ?{|
+      constKey?: string,
+      constVal?: string | number,
+    |},
+  ): void {
+    // 1) ensure the stylesheet exists
+    if (!this.injected) {
       this.inject();
     }
 
-    if (rawRTLRule != null) {
+    // 2) defineConsts update: patch or register constant values
+    if (
+      rawLTRRule === '' &&
+      metadata?.constKey != null &&
+      metadata.constVal != null
+    ) {
+      const key: string = (metadata.constKey: any);
+      const newVal: string = String((metadata.constVal: any));
+      console.log(`[defineConsts] invoked with key=${key}, constVal=${newVal}`);
+      console.log('[defineConsts] cache before:', this.constantsCache);
+
+      let entry = this.constantsCache.get(key);
+      if (!entry) {
+        // first defineConsts for this key nothing recorded yet
+        console.log(`[defineConsts] first define for key=${key}, recording value only`);
+        entry = { value: newVal, rules: [] };
+        this.constantsCache.set(key, entry);
```

Comment:
It would be slow, but you *might* want to search for the newly added `constKey` in *all* existing rules.

Since this primarily done for hot reloading, there is a possibility that someone would use a constant before defining it.